08-software-versions.Rmd

```diff
@@ -5,6 +5,108 @@
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g201bd406763_37_1492")
+```
+
+The `renv` workflow looks like this (as described by their documentation):
+
+> 1. Call `renv::init()` to initialize a new project-local environment with a private R library,
+
+> 2. Work in the project as normal, installing and removing new R packages as they are needed in the project,
+
+> 3. Call `renv::snapshot()` to save the state of the project library to the lockfile (called renv.lock),
+
+> 4. Continue working on your project, installing and updating R packages as needed.
+
+> 5. Call `renv::snapshot()` again to save the state of your project library if your attempts to update R packages were successful, or call `renv::restore()` to revert to the previous state as encoded in the lockfile if your attempts to update packages introduced some new problems.
+
+To make this shareable to others, you will need to do two things:
+
+1. Be sure to commit and push the `renv.lock` file to your GitHub repository for your project.
+2. Be sure to describe that your project uses `renv` in the README of this project (commit and push this to your GitHub repository also).
+
+```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "The renv workflow begins with intializing an environment with renv::init(). Then you install or remove packages and otherwise work in R as normal. When you are ready to update the renv snapshot, you run renv::snapshot(). You can now share this environment snapshot on GitHub or wherever. The environment can be restored using renv::restore()"}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g21dfe2f76f9_90_163")
+```
+
+The limitations of this method, [as noted by the `renv` authors](https://rstudio.github.io/renv/articles/renv.html#caveats), is that it really only tracks packages in R and cannot help track or enforce items that may affect the computing environment outside of R. So while it will aid in the reproducibility of your analysis, it will not cover everything.
+
+## Containerization
+
+In order to truly reproduce a result with an identical computing environment you would need to use a containerized approach. To containerize a computing environment is to truly create an environment that is shippable to others. A container is analogous to a virtual machine. A computer runs a computing environment inside of it that is separate from the rest of the computer (hence why its called a container).
```

Comment:
maybe define virtual machine too
maybe make a dictionary box about containerization (possible add virtual machine too it too)

Reply:
could also add Docker and Singualarity to that